python/global_search_widget/global_search_widget.py

```diff
@@ -0,0 +1,284 @@
+    def _clear_model(self, add_loading_item=True):
+        """
+        Clears the current data in the model.
+        
+        :param add_loading_item: if true, a "loading please wait" item will
+                                 be added.
+        """
+        # clear model
+        self._model.clear()
+        
+        if add_loading_item:
+            item = QtGui.QStandardItem("Loading data...")
+            item.setData(self.MODE_LOADING, self.MODE_ROLE)
+            self._model.appendRow(item)
+        
+    def _on_text_changed(self, text):
+        """
+        Callback that fires when the user types something into
+        the text editor.
+        
+        :param text: current contents of editor
+        """
+        if len(text) < 3:
+            # global search wont work with shorter than 3 len strings
+            # for these cases, clear the auto completer model fully
+            # there is no more work to do
+            self._clear_model(add_loading_item=False)
+            return
+        
+        # now we are about to populate the model with data 
+        # and therefore trigger the completer to pop up.
+        #
+        # The completer seems to have some internal properties
+        # which are transitory and won't last between sessions.
+        # for these, we have to set them up every time the 
+        # completion process is about to start it seems.
+        
+        # tell completer to render matches using our delegate
+        # configure how the popup should look
+        
+        # deferred import to help documentation generation.
+        from .search_result_delegate import SearchResultDelegate
+        
+        self._popup = self._completer.popup()
+        self._delegate = SearchResultDelegate(self._popup)
+        self._completer.popup().setItemDelegate(self._delegate)
+        
+        # try to disconnect and reconnect the activated signal
+        # it seems this signal is lost every time the widget
+        # looses focus.
+        try:
+            self._completer.activated[QtCore.QModelIndex].disconnect(self._on_completer_select)
+        except:
+            self._bundle.log_debug("Could not disconnect global search activated "
+                                "signal prior to reconnect. Looks like this connection "
+                                "must have been discarded at some point along the way.")
+                
+        self._completer.activated[QtCore.QModelIndex].connect(self._on_completer_select)
+        
+        # now clear the model
+        self._clear_model()
+
+        # clear download queue
+        self.__sg_data_retriever.clear()
+        
+        # clear thumbnail map
+        self._thumb_map = {}
+        
+        # kick off async data request from shotgun 
+        data = {"text": text}
+        self._processing_id = self.__sg_data_retriever.execute_method(self._do_sg_global_search, data)        
+        
+    def _do_sg_global_search(self, sg, data):
```

Comment:
Same question as note editor. Just wondering if `data` needs to be a dict. Is there other data that may go in here?